### PR TITLE
Tests/cypress

### PR DESCRIPTION
**Pull request by kaluger-roman**
- Cypress tests for main functionality was added
- Docker environment for cypress was created
- Github actions pipeline was added

### CODE REVIEW

File: packages/e2e/cypress/integration/e2e/Canvas/Canvas.model.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { RootModel, selectors } from "../../../models"
+
+export class CanvasModel extends RootModel {
+  canvasPosition() {
+    return this.getCanvas().then(($el) => $el.css("transform"))
```
Comment (kseniaSs) [inline packages/e2e/cypress/integration/e2e/Canvas/Canvas.model.ts]: Selectors to diffrent file

File: packages/e2e/cypress/integration/e2e/Canvas/canvas.cy.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      .canvasPositionOrigin()
+      .then(coordinatesFromStringPX)
+      .then(([xCoord, yCoord]) => {
+        expect(Number(xCoord)).to.be.closeTo(x, BROWSER_PX_DEVIATION)
+        expect(Number(yCoord)).to.be.closeTo(y, BROWSER_PX_DEVIATION)
+      })
+
+  describe("Initializing", () => {
+    it("Should match default position", () => {
+      checkCanvasPosition(15, -35)
```
Comment (kseniaSs) [inline packages/e2e/cypress/integration/e2e/Canvas/canvas.cy.ts]: Magical. numbers

File: packages/e2e/cypress/integration/e2e/Connections/connections.cy.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  beforeEach(connectionsModel.open)
+
+  const checkConnectionsCount = (count: number) =>
+    connectionsModel.getConnections().then(($cons) => expect($cons.length).to.equal(count))
+
+  describe("Connections logic", () => {
+    const verifyFirstConnectionInitial = () =>
+      connectionsModel
+        .getFirstConnectionPath()
+        .should("be.equal", "M 15310 15335 C 15210 15335, 15359 15140, 15259 15140")
```
Comment (kseniaSs) [inline packages/e2e/cypress/integration/e2e/Connections/connections.cy.ts:17]: Magical number and on the bottom too

File: packages/e2e/cypress/integration/e2e/Nodes/deletions.cy.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+context(CONTEXT, () => {
+  beforeEach(nodesModel.open)
+
+  const checkNodesCount = (count: number) =>
+    nodesModel.nodesElements().then(($el) => expect($el.get()).to.have.length(count))
+
+  describe("Deletions", () => {
+    it("Should not do anything without selected nodes", () => {
+      cy.realPress("Backspace")
```
Comment (kseniaSs) [inline packages/e2e/cypress/integration/e2e/Nodes/deletions.cy.ts]: It should be keyboard contants

File: packages/e2e/package.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "tests",
```
Comment (kseniaSs) [inline packages/e2e/package.json]: Why are tests in packages?I think it's wrong, see please different packages

File: packages/e2e/cypress/integration/e2e/Canvas/Canvas.model.ts (hunk 1 above)
Comment (kaluger-roman) [reply packages/e2e/cypress/integration/e2e/Canvas/Canvas.model.ts]: but it's css props names, not selectors, should we move it in constants too?

File: packages/e2e/cypress/integration/e2e/Canvas/canvas.cy.ts (hunk 2 above)
Comment (kaluger-roman) [reply packages/e2e/cypress/integration/e2e/Canvas/canvas.cy.ts]: ok, i've looked through tests and replaced a part of numbers to constants that has real meaningful meaning and is reused many times (i. e. nodes coords, connectors coords), but some other magic numbers have no it's meaningful meaning and is used 'in place' in tests, so, i suppose that part of numbers is redundant to be moved to constants

File: packages/e2e/cypress/integration/e2e/Canvas/Canvas.model.ts (hunk 1 above)
Comment (kseniaSs) [reply packages/e2e/cypress/integration/e2e/Canvas/Canvas.model.ts]: css styles Maybe or smth like this

File: packages/e2e/cypress/integration/e2e/Canvas/canvas.cy.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      canvasModel.dnd(550, 550, 550, CANVAS_MOVE_POINT_Y)
+      canvasModel.dnd(550, CANVAS_MOVE_POINT_Y, 450, 450)
+
+      matchEndPosition()
+    })
+
+    it("Should move canvas long movement", () => {
+      canvasModel
+        .mouseDown(CANVAS_MOVE_POINT_X, CANVAS_MOVE_POINT_Y)
+        .realMouseMove(550, 550)
```
Comment (kseniaSs) [inline packages/e2e/cypress/integration/e2e/Canvas/canvas.cy.ts]: For example, here. It's coordinates and it uses several times, i think it's should be separate

**Review by kseniaSs — APPROVED** [review]

**Pull request merged**